### ISSUE TITLE
Don't remap the start of the RHS if it matches the LHS

### ISSUE THREAD

**Issue by martinwguy** [martinwguy/xvi#48]
Classic vi allows
```
:se remap
:map! the there
athe
```
without going into an infinite loop. It seems to check if the start of the RHS matches the whole LHS and, if so, not remap the initial part of the RHS.

nvi and vim have the same behaviour
xvi goes into an infinite remapping loop.

**Comment by martinwguy** [comment]
POSIX confirms
"The rhs characters shall themselves be subject to remapping, unless otherwise specified by the remap edit option, except that if the characters in lhs occur as prefix characters in rhs, those characters shall not be remapped."
http://pubs.opengroup.org/onlinepubs/9699919799/utilities/ex.html#tag_20_40_13_24)

**Comment by martinwguy** [comment]
The POSIX standard does not reflect classic vi's behaviour.
'``
:map! the there
:map! h H
athe
```
gives "tHere" in classic vi and nvi, "there" in vim (and presumably POSIX)

**Comment by martinwguy** [comment]
POSIX behaviour implemented by 0c27e86d

**Comment by martinwguy** [comment]
POSIX behaviour implemented by 0c27e86d
